forte/processors/data_augment/algorithms/distribution_replacement_augmenter.py

```diff
@@ -0,0 +1,112 @@
+    from which sample by a given distribution to subsitute a word
+    from the original text. The distributions implemented now are uniform,
+    unigram (related to a specific dataset).
+    Todo: More distriubtion types can be added.
+    """
+    def __init__(self, configs: Dict[str, str]):
+        random.seed(0)
+        super().__init__(configs)
+        try:
+            self.vocab = set(words.words())
+            self.stop_words = set(stopwords.words(self.configs['lang']))
+        except LookupError:
+            nltk.download('words')
+            nltk.download('stopwords')
+            self.vocab = set(words.words('en'))
+            self.stop_words = set(stopwords.words(self.configs['lang']))
+
+    @property
+    def replacement_level(self) -> List[str]:
+        return ["word"]
+
+    def _uniform_sample(self) -> str:
+        """
+        return a random word from vocab
+        """
+        word = random.choice(self.vocab - self.stop_words)
+        return word
+
+    def _unigram_sample(self) -> str:
+        """
+        Load the unigram distribution of one dataset from file and
+        return a word sampled from unigram distribution
+        """
+        vocab, cnt = [], []
+        with open(self.configs['unigram'], 'r') as file:
+            for line in file:
+                if len(line) > 0:
+                    line = line.strip()
+                    word, count = line.split(' ')
+                    vocab.append(word)
+                    cnt.append(int(count))
+        return random.choices(vocab, cnt)[0]
+
+    def augment(self, word: str) -> str:
+        r"""
+        This function replaces a word with synonyms from a WORDNET dictionary.
+        Args:
+            word: input
+            additional_info: contains pos_tag of the word, optional
+        Returns:
+            a synonym of the word
+        """
+        if self.configs['distribution'] == "uniform":
+            return self._uniform_sample()
+        elif self.configs['distribution'] == "unigram":
+            if self.configs['unigram'] == "":
+                raise KeyError("Unigram distribution is not provided in config.")
+            else:
+                return self._unigram_sample()
+        else:
+            raise NotImplementedError
+
+    def default_configs(cls):
+        """
+        :return: A dictionary with the default config for this processor.
+        Following are the keys for this dictionary:
+            - distribution: defines what types of distribution to be used.
+                Options are uniform, unigram, etc.
+            - unigram: An input file with unigram stats of the specific dataset.
+                if distribution type is set to "unigram" and "unigram" is not provided,
+                then simply use _generate_unigram function to generate one
+                from the given text.
+        """
+        config = super().default_configs()
+        config.update({
```

Comment:
So the better design is to pass in a "Sampler", and this class don't need to understand the Sampler type, it only knows that it can draw samples from the "Sampler".